packages/decoupled/src/cache/fetch.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * Function to query cached Fetch state
+ */
+
+import { genAPICacheKey } from '../lib';
+import FetchStore from './fetch-store';
+import { Site } from '../site/site';
+
+export default async (site: Site, fetch: CallableFunction, { type, params }): Promise<AnyObject> => {
```

Comment:
see above, I think this is solution id not enough to decouple the fetch, cache and the data-source handler.